pymc3/stats.py

```diff
@@ -232,6 +237,71 @@ def quantiles(x, qlist=(2.5, 25, 50, 75, 97.5)):
         print("Too few elements for quantile calculation")
 
 
+def df_summary(trace, varnames=None, stat_funcs=None, alpha=0.05, batches=100):
+    """Create a data frame with summary statistics.
+
+    Parameters
+    ----------
+    trace : MultiTrace instance
+    varnames : list
+        Names of variables to include in summary
+    stat_funcs : None or list
+        A list of functions used to calculate statistics. By default,
+        the mean, standard deviation, simulation standard error, and
+        highest posterior density intervals are included.
+
+        The functions will be given one argument, the samples for a
+        variable as a 2 dimensional array, where the first axis
+        correspond to sampling iterations and the second axis
+        represent the flattened variable (e.g., x__0, x__1,...). It
+        should return either
+        1) a Pandas Series instance where each value corresponds to a
+           column of the passed in array and the Series name is the
+           name of the statistic
+        2) or a Pandas DataFrame instance where the rows correspond to
+           the columns of the passed in array and the column names are
+           the names of the statistics.
+    alpha : float
+        The alpha level for generating posterior intervals. Defaults
+        to 0.05. This is only meaningful when `stat_funcs` is None.
+    batches : int
+      Batch size for calculating standard deviation for
+      non-independent samples. Defaults to 100. This is only
+      meaningful if `stat_funcs` when None.
+
+    See also
+    --------
+    summary : Generate a pretty-printed summary of a trace.
+
+    Returns
+    -------
+    Data frame with summary statistics for each variable
+    """
+    if varnames is None:
+        varnames = trace.varnames
+
+    if stat_funcs is None:
+        stat_funcs = [lambda x: pd.Series(np.mean(x, 0), name='mean_'),
```

Comment:
> I suppose this is done so as not to collide with the `mean()` method?

Yep.

> Note that pandas will would still call `mean()` if you do `df.mean()`
> even with such a column present, but evaluate the column if you did
> `df.mean`.

Is that a feature of a recent pandas?  With the version of pandas I have
at hand (0.16.2), the method call will still work, but trying to select
the column with `.mean` does not (you instead have to use `[mean]`).